scripts/components/Search.js

```diff
@@ -13,11 +13,17 @@ export default class Search extends React.Component {
 
     this.state = {
       query: '',
+      hasTyped: false,
     }
   }
 
   handleQueryChange = (event) => {
     this.setState({ query: event.currentTarget.value })
+
+    // if the query is not empty, the user has typed at least once
+    // this might be better elsewhere
+    if (event.currentTarget.value !== '')
+      this.setState({ hasTyped: true })
```

Comment:
You might be able to move this into the first `setState` and retain the same behavior.